src/containers/BooksForm.jsx

```diff
@@ -1,27 +1,80 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import { categories } from '../store';
+import { createBook } from '../actions';
 
-const BooksForm = () => (
-  <div>
-    <h3>Add New Book</h3>
-    <form>
-      <div>
-        <label htmlFor="title">
-          Title:
-          <input type="text" id="title" placeholder="Book Title" />
-        </label>
-      </div>
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    const { createBook } = this.props;
+    const { title, category } = this.state;
+    const book = {
+      id: Math.floor(Math.random() * 100),
+      title,
+      category,
+    };
+
+    if (title && category) {
+      createBook(book);
+      this.setState({
+        title: '',
+        category: '',
+      });
+    }
+
+    event.preventDefault();
+  }
```

Comment:
- [x] Let's add the `event.preventDefault()` to the first line. Adding it to the last line opens the possibility that the interpreter will not reach the last line and the browser will proceed to execute the button's default behavior.

- [x] Instead of using `if (title && category)`, let's add validation to the select and input tags.

```suggestion
  handleSubmit(event) {
    event.preventDefault();
    const { createBook } = this.props;
    const { title, category } = this.state;
    const book = {
      id: Math.floor(Math.random() * 100),
      title,
      category,
    };
    createBook(book);
    this.setState({
      title: '',
      category: '',
    });
  }
```